lib/datadog/core/configuration/option.rb

```diff
@@ -102,6 +191,33 @@ def context_eval(&block)
           @context.instance_eval(&block)
         end
 
+        def set_value_from_env_or_default
+          value = nil
+          precedence = nil
+
+          if definition.env && ENV[definition.env]
+            value = coerce_env_variable(ENV[definition.env])
+            precedence = Precedence::PROGRAMMATIC
+          end
+
+          if value.nil? && definition.deprecated_env && ENV[definition.deprecated_env]
+            value = coerce_env_variable(ENV[definition.deprecated_env])
+            precedence = Precedence::PROGRAMMATIC
+
+            Datadog::Core.log_deprecation do
+              "#{definition.deprecated_env} environment variable is deprecated, use #{definition.env} instead."
+            end
+          end
+
+          option_value = value.nil? ? default_value : value
+
+          set(option_value, precedence: precedence || Precedence::DEFAULT)
+        end
+
+        def skip_validation?
+          !!ENV['DD_EXPERIMENTAL_SKIP_CONFIGURATION_VALIDATION']
```

Comment:
To make sure that we don't trigger unless it's being set to the correct thing:

```suggestion
          ['true', '1'].include?(ENV.fetch('DD_EXPERIMENTAL_SKIP_CONFIGURATION_VALIDATION', '').strip)
```